core/sdk-core/src/main/java/software/amazon/awssdk/core/waiters/WaitersRuntime.java

```diff
@@ -0,0 +1,523 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.waiters;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.core.SdkPojo;
+import software.amazon.awssdk.core.SdkResponse;
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.core.exception.SdkServiceException;
+import software.amazon.awssdk.utils.ToString;
+
+/**
+ * Contains classes used at runtime by the code generator classes for waiter acceptors generated from JMESPath expressions.
+ */
+@SdkProtectedApi
+public final class WaitersRuntime {
```

Comment:
Can we add tests for the runtime?

Reply:
That's a good idea, but I'm wondering how that will jive when we move the runtime to being copied into each of the service clients. I was hoping testing it in the context it's used (i.e. testing the generated waiters and making sure they worked, which indirectly tests the runtime) would be sufficient to side-step this problem. Do you have any ideas on how to make that work?

Reply:
> Do you have any ideas on how to make that work?

I think if we just accessed a `WaitersRuntime` class from within `codegen-generated-classes-test` that could work.

Reply:
Cool, so generate it into each service as well? That could work...